core/shared/utils/bh_atomic.h

```diff
@@ -69,7 +107,16 @@ typedef uint16 bh_atomic_16_t;
     __atomic_fetch_add(&(v), (val), __ATOMIC_SEQ_CST)
 #define BH_ATOMIC_32_FETCH_SUB(v, val) \
     __atomic_fetch_sub(&(v), (val), __ATOMIC_SEQ_CST)
+#else /* else of BH_ATOMIC_32_IS_ATOMIC == 1 */
+#define BH_ATOMIC_32_LOAD(v) (v)
+#define BH_ATOMIC_32_STORE(v, val) (v) = val
+#define BH_ATOMIC_32_FETCH_OR(v, val) nonatomic_32_fetch_or(&(v), val)
+#define BH_ATOMIC_32_FETCH_AND(v, val) nonatomic_32_fetch_and(&(v), val)
+#define BH_ATOMIC_32_FETCH_ADD(v, val) nonatomic_32_fetch_add(&(v), val)
+#define BH_ATOMIC_32_FETCH_SUB(v, val) nonatomic_32_fetch_sub(&(v), val)
```

Comment:
Should move the below implementations of nonatomic_32_fetch_or/nonatomic_32_fetch_and/nonatomic_32_fetch_add/nonatomic_32_fetch_sub here, they should not be wrapped by `BH_ATOMIC_16_IS_ATOMIC`?

Reply:
Sure